VMBackup/main/handle.py

```diff
@@ -603,20 +614,22 @@ def enable():
         para_parser = ParameterParser(protected_settings, public_settings, backup_logger)
 
         if(para_parser.taskId is not None and para_parser.taskId != ""):
+            eventlogger.update_properties(para_parser.taskId)
+            hutil.set_event_logger(eventlogger)
             backup_logger.log('taskId: ' + str(para_parser.taskId), True)
             randomSleepTime = random.randint(500, 5000)
             backup_logger.log('Sleeping for milliseconds: ' + str(randomSleepTime), True)
             time.sleep(randomSleepTime / 1000)
             exit_if_same_taskId(para_parser.taskId)
             taskIdentity = TaskIdentity()
-            taskIdentity.save_identity(para_parser.taskId)
-
+            taskIdentity.save_identity(para_parser.taskId)       
         temp_status= 'success'
         temp_result=CommonVariables.ExtensionTempTerminalState
         temp_msg='Transitioning state in extension'
         blob_report_msg, file_report_msg = get_status_to_report(temp_status, temp_result, temp_msg, None)
 
         status_report_to_file(file_report_msg)
+        eventlogger.dispose()
```

Comment:
what is the idea behind not having one event logger (both for daemon and enable)? Aren't we following singleton pattern?